test/index.js

```diff
@@ -162,477 +140,262 @@ describe('koa-body', () => {
-
-      request(http.createServer(app.callback()))
-        .post('/echo_body')
-        .type('text')
-        .send('plain text content')
-        .expect(200)
-        .end((err, res) => {
-          if (err) return done(err);
-
-          assert(requestSpy.calledOnce, 'Spy for /echo_body not called');
-          const req = requestSpy.firstCall.args[0].request;
-          should(req.body).equal('plain text content');
-
-          // Raw text requests are still just text
-          assert.equal(req.body[unparsed], undefined);
-
-          // Text response is just text
-          should(res.body).have.properties({});
-          should(res.text).equal('plain text content');
-
-          done();
-        });
-    });
-
-  });
-
   /**
    * TEXT request body
    */
-  it('should recieve `text` request bodies',  (done) => {
+  it('should recieve `text` request bodies', async () => {
     app.use(koaBody({ multipart: true }));
     app.use(router.routes());
 
-    request(http.createServer(app.callback()))
+    return request(http.createServer(app.callback()))
       .post('/echo_body')
       .type('text')
       .send('plain text')
       .expect(200)
-      .end( (err, res) => {
-        if (err) return done(err);
-
+      .then((res) => {
         res.type.should.equal('text/plain');
         res.text.should.equal('plain text');
-
-        done();
       });
   });
 
-  describe('strict mode',  () => {
-    beforeEach( () => {
-      //push an additional, to test the multi query
+  describe('strict mode', async () => {
+    beforeEach(async () => {
+      // push an additional, to test the multi query
       database.users.push({ name: 'charlike' });
     });
 
-    it('can enable strict mode',  (done) => {
+    it('can enable strict mode', async () => {
       app.use(koaBody({ strict: true }));
       app.use(router.routes());
 
-      request(http.createServer(app.callback()))
+      return request(http.createServer(app.callback()))
         .delete('/users/charlike')
         .type('application/x-www-form-urlencoded')
         .send({ multi: true })
         .expect(204)
-        .end( (err, res) => {
-          if (err) return done(err);
+        .then(() => {
           assert(database.users.find(element => element.name === 'charlike') !== undefined);
-          done();
         });
     });
 
-    it('can disable strict mode',  (done) => {
+    it('can disable strict mode', async () => {
       app.use(koaBody({ strict: false }));
       app.use(router.routes());
 
-      request(http.createServer(app.callback()))
+      return request(http.createServer(app.callback()))
         .delete('/users/charlike')
         .type('application/x-www-form-urlencoded')
         .send({ multi: true })
         .expect(204)
-        .end( (err, res) => {
-          if (err) return done(err);
+        .then(() => {
           assert(database.users.find(element => element.name === 'charlike') === undefined);
```

Comment:
Third testing style `.should`, `should( )` and `assert( )` are all present now. Just putting this up for discussion

Reply:
good catch, I'm trying to purge `assert` from the codebase

Reply:
we're down to just .should and should( ) now